feat(parameters): migrate AppConfig to new APIs due to API deprecation (#1553)
awslabs/aws-lambda-powertools-python: docs/upgrade.md

* The API for **event handler's `Response`** has minor changes to support multi value headers and cookies.
* The **legacy SQS batch processor** was removed.
* The **Idempotency key** format changed slightly, invalidating all the existing cached results.

> * The **Feature Flags and AppConfig Parameter utility** API calls have changed and you must add new IAM permissions to Lambda.

Reviewer: ```suggestion
* The **Feature Flags and AppConfig Parameter utility** API calls have changed and you must update your IAM permissions.
```

After this change, the key is generated using the `module name` + `qualified function name` + `idempotency key` (e.g: `app.classExample.function#app.handler#282e83393862a613b612c00283fef4c8`).

Using qualified names prevents distinct functions with the same name to contend for the same Idempotency key.

> ## Feature Flags and AppConfig Parameter utility

Author: Hi @heitorlessa! I need help to improve the documentation. Maybe adding best practices using all parameters under the same configuration?

Reviewer: Sure. We should add an `Info` Admonition in the Parameters and Feature Flags docs.

The current API is [deprecated](https://docs.aws.amazon.com/appconfig/2019-10-09/APIReference/API_GetConfiguration.html) for new accounts and we had to change this.

> No changes are required to your code, but you must add `appconfig:GetLatestConfiguration` and `appconfig:StartConfigurationSession` IAM permissions for this to work in your Lambdas.

Reviewer: ```suggestion
AWS AppConfig deprecated the current API (GetConfiguration) - [more details here](https://github.com/awslabs/aws-lambda-powertools-python/issues/1506#issuecomment-1266645884).

You must update your IAM permissions to allow `appconfig:GetLatestConfiguration` and `appconfig:StartConfigurationSession`. There are no code changes required.
```
